modal/image.py

```diff
@@ -2389,5 +2390,15 @@ async def _logs(self) -> typing.AsyncGenerator[str, None]:
                 if task_log.data:
                     yield task_log.data
 
+    async def hydrate(self, client: Optional[_Client] = None) -> Self:
+        """mdmd:hidden"""
+        # Image inherits hydrate() from Object but can't be hydrated on demand
+        # Overriding the method lets us hide it from the docs and raise a better error message
+        if not self.is_hydrated:
+            raise ExecutionError(
+                "Images cannot currently be hydrated on demand; you can build an Image by running an App that uses it."
+            )
+        return self
```

Comment:
Is it safer to call `super().hydrate`, catch the exception and then re-raise with a more specific exception?

Reply:
Yea it's the status quote, I think all use cases calls `_hydrate` properly.

The only weird case is `Image._log`:

https://github.com/modal-labs/modal-client/blob/c9c4ee845d0df1c19ac808b39b930f09af454c70/modal/image.py#L2278-L2283

An App needs to hydrate the `image` object before `_logs` can be used, but that already breaks today.